statsmodels/tsa/automatic/tests/test_sarimax.py

```diff
@@ -0,0 +1,24 @@
+"""unit test for automatic sarimax forecasting."""
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+
+# # Dataset
+wpi1 = requests.get('http://www.stata-press.com/data/r12/wpi1.dta').content
+data = pd.read_stata(BytesIO(wpi1))
+data.index = data.t
+
+def auto_order(endog, criteria='aic', d=0, max_order=(3, 3), spec=None):
```

Comment:
You don't need to (and shouldn't) replicate the function to be tested in the unit test. Instead, you should import the function you want to test, e.g.:

```python
from statsmodels.tsa.automatic import sarimax
```

and then in your `test_non_stepwise` function you should do something like:

```python
p, q = sarimax.auto_order(...)  # where you fill in the appropriate parameters for your test
desired_p = ...  # where you fill in whatever you think the returned `p` should be equal to
desired_q = ...  # similarly for what the returned `q` should be

assert_equal(p, desired_p)
assert_equal(q, desired_q)
```

In general, you'll want to test out various different types of parameterizations (e.g. different `max_order`, different `spec`, etc).